### PR TITLE
 scala 2.13 cross compile support for the 5.6.x branch

### PR DESCRIPTION
**Pull request by samzilverberg**
added support for scala 2.13 for the 5.6.x release.

did:

- started from release/5.6.x
- Build.scala 
  - added 2.13.1 to crossScalaVersions
  - update dependencies to ones that support 2.11 till 2.13
- compiled and fixed errors where needed
- ran tests on all cross scala versions

### CODE REVIEW

File: elastic4s-http-streams/src/main/scala/com/sksamuel/elastic4s/streams/ScrollPublisher.scala [hunk 1]
```diff
@@ -150,7 +150,7 @@ class PublishActor(client: HttpClient,
     // more results and we can unleash the beast (stashed requests) and switch back to ready mode
     case Success(resp: SearchResponse) =>
       scrollId = resp.scrollId.getOrElse("Query didn't return a scroll id")
-      queue.enqueue(resp.hits.hits: _*)
+      resp.hits.hits.foreach(h => queue.enqueue(h))
```
Comment (samzilverberg) [inline elastic4s-http-streams/src/main/scala/com/sksamuel/elastic4s/streams/ScrollPublisher.scala:153]: could be niced with `queue.enqueueAll` and `scala-collection-compat` lib.
I tried it but its missing support for `mutable.Queue.enqueueAll`.
I added it in a PR to the compat lib, but might take a couple of weeks until it appears in a release.

File: project/Build.scala [hunk 2]
```diff
@@ -11,22 +11,22 @@ object Build extends AutoPlugin {
 
   object autoImport {
     val org = "com.sksamuel.elastic4s"
-    val AkkaVersion = "2.4.17"
-    val CatsVersion = "0.9.0"
-    val CirceVersion = "0.7.1"
+    val AkkaVersion = "2.5.27"
+    val CatsVersion = "2.0.0"
+    val CirceVersion = "0.12.0-M3"
```
Comment (samzilverberg) [inline project/Build.scala:16]: another pr adding 2.13 support to 6.5 handles this differently:
https://github.com/sksamuel/elastic4s/pull/1874/files (search for circeversion)

I could try doing the same if you think its important.

Comment (sksamuel) [reply project/Build.scala:16]: This is fine.

**Comment by sksamuel** [comment]
I've released 5.611

**Pull request merged**